students/njschafi/Lesson03/assignment/test_basic_operations.py

```diff
@@ -0,0 +1,115 @@
+    def test_add_customer(self):
+        """Tests if a new customer is added to database"""
+        create_empty_db()
+        add_customer(**user_1)
+        query = Customer.get(Customer.customer_id == user_1['customer_id'])
+        self.assertEqual(user_1['name'], query.customer_name)
+        self.assertEqual(user_1['lastname'], query.customer_last_name)
+        self.assertEqual(user_1['home_address'], query.customer_address)
+        self.assertEqual(user_1['phone_number'], query.customer_phone)
+        self.assertEqual(user_1['email_address'], query.customer_email)
+        self.assertEqual(user_1['status'], query.customer_status)
+        self.assertEqual(user_1['credit_limit'], query.customer_limit)
+
+        # add another person
+        add_customer(**user_2)
+        query = Customer.get(Customer.customer_id == user_2['customer_id'])
+        self.assertEqual(user_2['name'], query.customer_name)
+        self.assertEqual(user_2['lastname'], query.customer_last_name)
+        self.assertEqual(user_2['home_address'], query.customer_address)
+        self.assertEqual(user_2['phone_number'], query.customer_phone)
+        self.assertEqual(user_2['email_address'], query.customer_email)
+        self.assertEqual(user_2['status'], query.customer_status)
+        self.assertEqual(user_2['credit_limit'], query.customer_limit)
+
+        # add a duplicate person
+        self.assertRaises(ValueError, add_customer(**user_2))
+        drop_db()
+
+    def test_search_customer(self):
+        """Tests customer search function"""
+        create_empty_db()
+        add_customer(**user_1)
+        test_map = {'name': user_1['name'], 'lastname': user_1['lastname'],
+                    'email': user_1['email_address'],
+                    'phone_number': user_1['phone_number']}
+        self.assertEqual(test_map, search_customer(user_1['customer_id']))
+
+        # Non-existant Customer Test
+        self.assertEqual({}, search_customer('99999'))
+        drop_db()
+
+    def test_delete_customer(self):
+        """Tests if user can delete customer(s)"""
+        create_empty_db()
+        add_customer(**user_1)
+        delete_customer(user_1['customer_id'])
+        self.assertEqual({}, search_customer(user_1['customer_id']))
+        drop_db()
+
+    def test_update_customer_credit(self):
+        """Tests if customer credit is updated properly"""
+        create_empty_db()
+        add_customer(**user_1)
+        update_customer_credit(user_1['customer_id'], 5000.00)
+        query = Customer.get(Customer.customer_id == user_1['customer_id'])
+        self.assertEqual(5000.00, query.customer_limit)
+
+        # Test for non-existant customer
+        update_customer_credit('456879', 5000.00)
+        self.assertRaises(ValueError,
+                          update_customer_credit('456879', 5000.00))
+
+        # Test for non-float value inputted
+        self.assertRaises(TypeError,

```

Comment:
Same issue here, the exception will cause the test to be reported as failing (not the case, since the exception is indeed raised). Change it to a context manager.